packages/wdio-browserstack-service/src/reporter.ts

```diff
@@ -50,7 +50,7 @@ class _TestReporter extends WDIOReporter {
     }
 
     private getUserCaps(runnerStats: RunnerStats) {
-        return runnerStats.instanceOptions[runnerStats.sessionId].capabilities
+        return runnerStats.instanceOptions[runnerStats.sessionId]?.capabilities
```

Comment:
Can we add null check for instanceOptions too?

Reply:
According to the defintion of [RunnerStats type](https://github.com/webdriverio/webdriverio/blob/6e8c6f8013df7a9abf4be69934a9e88ba2dc6268/packages/wdio-reporter/src/stats/runner.ts#L10), instanceOptions is never null, so that won't be needed.

Even if it is null, this won't break anything, as there is already a implicit try catch in this method

Reply:
There is already a [safety check](https://github.com/sriteja777/webdriverio/blob/15eb207ac49f731f52a485defcff91cf3c23de02/packages/wdio-browserstack-service/src/reporter.ts#L305) in place (It wraps every method of class with error handler). Even if this method fails, it doesn't effect any of the products.